static/js/containers/PlotContainer.jsx

```diff
@@ -1,12 +1,70 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 
 import Plot from '../components/Plot';
-import { FETCH_SOURCE_PLOT } from '../actions';
-import * as API from '../API';
+import * as Actions from '../actions';
 
-const fetchPlotData = url => (
-  API.GET(url, FETCH_SOURCE_PLOT)
+
+class PlotContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: false,
+      fetchingPlotIDs: []
+    };
+  }
+  componentWillMount() {
+    this.fetchPlotDataIfNotCached();
+  }
+  componentWillReceiveProps() {
+    this.fetchPlotDataIfNotCached();
+  }
+  fetchPlotDataIfNotCached() {
+    if (!this.props.plots.plotIDList.includes(this.props.url) &&
+        !this.state.fetchingPlotIDs.includes(this.props.url)) {
+      this.props.dispatch(
+        Actions.fetchPlotData(
+          this.props.url,
+          Actions.FETCH_SOURCE_PLOT
+        )
+      );
+      this.setState(
+        { fetchingPlotIDs: this.state.fetchingPlotIDs.concat([this.props.url]) }
+      );
+    }
+    if (this.props.plots.plotData[this.props.url] &&
+        this.state.fetchingPlotIDs.includes(this.props.url)) {
+      const fetchingPlotIDs = this.state.fetchingPlotIDs.slice();
```

Comment:
This is weird; if we really can't find a cleaner way to express this, a comment may be helpful.

Reply:
`splice` simply removes the arrray element(s) at the specified index. Does knowing that clarify this enough?

Reply:
Yes, I think a comment could help here, since `slice` has no argument.